src/components/steps/step_horizontal.styles.ts

```diff
@@ -28,6 +29,22 @@ export const euiStepHorizontalStyles = (euiThemeContext: UseEuiTheme) => {
    *    e.g. due to some of their titles wrapping to multiple lines
    */
 
+  const _generateStepSizeAndInset = (size: EuiStepsHorizontalSizes) => {
+    const stepNumberSize =
+      size === 's' ? euiStep.numberXSSize : euiStep.numberSize;
```

Comment:
Code suggestion: rather than making this take a `'s'` or `'m'` generator, change it to accept the actual number instead, i.e.
```suggestion
  const _generateStepSizeAndInset = (stepNumberSize: number) => {
```

This approach is more flexible in terms of supporting more sizes down the road in the future - e.g. if we add an `l` size,  we don't need to convert this ternary to a nested ternary or a switch, we can simply change what gets passed by each size prop, e.g. `s: _generateStepSizeAndInset(euiStep.numberLSize)`

Reply:
That totally makes sense! Thanks for the feedback!